### PR TITLE
fix(docs-infra): fix code hover style in TOC

### PR DESCRIPTION
**Pull request by sjtrimble**
Fixes code hover color to match text in table of contents per issue #31170

<hr>
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?
Issue Number: #31170

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by gkalpak — APPROVED** [review]
👍 

Please add `Fixes #31170` at the bottom of the commit message (per the [guidelines][1] :wink:)

[1]: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit

File: aio/src/styles/2-modules/_toc.scss [hunk 1]
```diff
@@ -144,12 +144,14 @@ aio-toc.embedded {
         display: table-cell;
       }
 
-      &:hover a {
-        color: $accentblue;
+      &:hover {
+        a, code {
```
Comment (gkalpak) [inline aio/src/styles/2-modules/_toc.scss]: Actually, is there any reason to limit this to `a` and `code`? Why not any element?

Comment (sjtrimble) [reply aio/src/styles/2-modules/_toc.scss]: Great point :) I'll update, that's cleaner and safer 👍

**Comment by gkalpak** [comment]
@sjtrimble, can you, please, update the commit message (see https://github.com/angular/angular/pull/31173#pullrequestreview-252505100)?

**Comment by sjtrimble** [comment]
> @sjtrimble, can you, please, update the commit message (see [#31173 (review)](https://github.com/angular/angular/pull/31173#pullrequestreview-252505100))?

@gkalpak Added the note to the commit message :) and rebased.

File: aio/src/styles/2-modules/_toc.scss [hunk 2]
```diff
@@ -144,12 +144,14 @@ aio-toc.embedded {
         display: table-cell;
       }
 
-      &:hover a {
-        color: $accentblue;
+      &:hover *{
```
Comment (gkalpak) [inline aio/src/styles/2-modules/_toc.scss]: Missing space after `*` 😱

File: aio/src/styles/2-modules/_toc.scss [hunk 3]
```diff
@@ -144,12 +144,14 @@ aio-toc.embedded {
...
+        // a, code {
```
Comment (gkalpak) [inline aio/src/styles/2-modules/_toc.scss]: Ahem...

File: aio/src/styles/2-modules/_toc.scss [hunk 4]
```diff
@@ -144,12 +144,14 @@ aio-toc.embedded {
...
+          color: $accentblue;
+        // }
       }
 
       &.active {
-        a {
+        a, code {
```
Comment (gkalpak) [inline aio/src/styles/2-modules/_toc.scss]: Use `*` here too?

**Review by gkalpak — CHANGES_REQUESTED** [review]

File: aio/src/styles/2-modules/_toc.scss (hunk 4 above)
Comment (sjtrimble) [reply aio/src/styles/2-modules/_toc.scss]: Yeah, good call 👍

File: aio/src/styles/2-modules/_toc.scss [hunk 5]
```diff
@@ -144,12 +144,12 @@ aio-toc.embedded {
         display: table-cell;
       }
 
-      &:hover a {
-        color: $accentblue;
+      &:hover * {
```
Comment (gkalpak) [inline aio/src/styles/2-modules/_toc.scss]: So why `&:hover *` but `&.active { *`? :thinking:

File: aio/src/styles/2-modules/_toc.scss [hunk 6]
```diff
@@ -144,12 +144,12 @@ aio-toc.embedded {
...
+          color: $accentblue;
```
Comment (gkalpak) [inline aio/src/styles/2-modules/_toc.scss]: Incorrect indentation (unless you switch to `&:hover { * { ...` :wink:).

**Review by gkalpak — APPROVED** [review]

File: aio/src/styles/2-modules/_toc.scss (hunk 6 above)
Comment (sjtrimble) [reply aio/src/styles/2-modules/_toc.scss]: Ok! I think we got it this time - made them both consistent. Also did some clean up while I was in there :)

Comment (gkalpak) [reply aio/src/styles/2-modules/_toc.scss]: 👍 

(Pro tip (for the future): It makes is much easier to review if clean-ups are on a separate commit :wink:)

File: aio/src/styles/2-modules/_toc.scss [hunk 7]
```diff
@@ -6,200 +6,203 @@
...
+          color: lighten($darkgray, 10);
+          overflow: visible;
+          @include font-size(12);
+          display: table-cell;
+        }
+
+        &:hover {
+          * {
+            color: $accentblue;
+            }
```
Comment (gkalpak) [inline aio/src/styles/2-modules/_toc.scss]: Incorrect indentation :smiley:

File: aio/src/styles/2-modules/_toc.scss [hunk 8]
```diff
@@ -6,200 +6,203 @@
...
+      }
     }
 
-    &:hover:not(.embedded) {
-      color: $accentblue;
+    button.toc-heading,
+    button.toc-more-items {
+      cursor: pointer;
+      display: inline-block;
+      background: 0;
```
Comment (gkalpak) [inline aio/src/styles/2-modules/_toc.scss]: Since you are cleaning up: What is this for? 🤔

**Review by gkalpak — APPROVED** [review]
👏

**Pull request closed without merging**